pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2292 @@
+        This step is for initialization.
+        3. Unfix the experiment design decisions and solve the two-stage DOE problem.
+        '''
+        time0 = time.time()
+
+        # store inputs in object
+        self.design_values = design_values
+        self.optimize = if_optimize
+        self.objective_option = objective_option
+        self.scale_nominal_param_value = scale_nominal_param_value
+        self.scale_constant_value = scale_constant_value
+        self.Cholesky_option = if_Cholesky
+        self.L_LB = L_LB
+        self.L_initial = L_initial
+        self.formula = formula
+        self.step = step
+        self.tee_opt = True
+
+        # calculate how much the FIM element is scaled by a constant number
+        # FIM = Jacobian.T@Jacobian, the FIM is scaled by squared value the Jacobian is scaled
+        self.fim_scale_constant_value = self.scale_constant_value **2
+
+        # check if inputs are valid
+        # simultaneous mode does not need to check mode and dimension of design variables
+        if check:
+            self.__check_inputs(check_mode=False)
+
+        # build the large DOE pyomo model
+        m = self.__create_doe_model()
+
+        # solve model, achieve results for square problem, and results for optimization problem
+
+        # Solve square problem first
+        # result_square: solver result
+        time0_solve = time.time()
+        result_square = self.__solve_doe(m, fix=True)
+        time1_solve = time.time()
+
+        time_solve1 = time1_solve-time0_solve
+
+        analysis_square = FIM_result(self.param_name, prior_FIM=self.prior_FIM, scale_constant_value=self.scale_constant_value)
+        analysis_square.extract_FIM(m, self.design_timeset, result_square, obj=objective_option)
+
+
+        if self.optimize:
+            # solve problem with DOF then
+            time0_solve2 = time.time()
+            result_doe = self.__solve_doe(m, fix=False)
+            time1_solve2 = time.time()
+
+            time_solve2 = time1_solve2 - time0_solve2
+
+            analysis_optimize = FIM_result(self.param_name, prior_FIM=self.prior_FIM)
+            analysis_optimize.extract_FIM(m, self.design_timeset, result_doe, obj=objective_option)
+            analysis_optimize.model = m
+
+            time1 = time.time()
+            if self.verbose:
+                print('Total solve time with simultaneous_finite mode (Wall clock) [s]:', time_solve1 + time_solve2)
+                print('Total wall clock time [s]:', time1-time0)
+
+            return analysis_square, analysis_optimize
+
+        else:
+            analysis_optimize.model = m
+
+            time1 = time.time()
+            if self.verbose:
+                print('Total solve time with simultaneous_finite mode (Wall clock) [s]:', time_solve1)
+                print('Total wall clock time [s]:', time1 - time0)
+
+            return analysis_square
```

Comment:
Should we be saving the timing data at returning it?

Reply:
For now I add these timing data to the result object